scripts/db-heal-test.py

```diff
@@ -0,0 +1,231 @@
+import sys
+import flag
+
+# --- Dynamic variables --- #
+aida_path_flag = flag.string("aida", "", "Path to Aida root.")
+aida_db_path_flag = flag.string("aida_db", "", "Path to AidaDB.")
+tmp_path_flag = flag.string("tmp", "", "Path to tmp dir.")
+
+number_of_iterations_flag = flag.int("iter", 1000, "Number of iterations.")
+window_flag = flag.int("window", 120, "How often will program get terminated (in seconds).")
+checkpoint_granularity_flag = flag.int("cp_granularity", 2000, "How often will Carmen create checkpoints.")
+flag.parse()
+
+aida_path = aida_path_flag.value
+aida_db_path = aida_db_path_flag.value
+tmp_path = tmp_path_flag.value
+
+number_of_iterations = number_of_iterations_flag.value
+window = window_flag.value
+checkpoint_granularity = checkpoint_granularity_flag.value
+
+# Mark first checkpoint
+latest_checkpoint = checkpoint_granularity
+
+if aida_path == "":
+    print("please set Aida using --aida")
+    exit(1)
+if aida_db_path == "":
+    print("please set AidaDB using --aidadb")
+    exit(1)
+if tmp_path == "":
+    tmp_path = tempfile.gettempdir()
+    print(f"tmp not set - using default {tmp_path}")
+
+# Block variables
+first_block = 0
+last_block = 60000000
+
+# --- Static variables --- #
+
+carmen_root = os.path.abspath('../go')
+
+# --- Script --- #
+
+# Log file path from which we read output to find kill_block
+aida_log_file = Path.cwd() / 'aida.log'
+carmen_log_file = Path.cwd() / 'carmen.log'
+current_dir = Path.cwd()
+
+print("Your settings:")
+print(f"\tNumber of iterations: {number_of_iterations}.")
+print(f"\tWindow size: {window} seconds.")
+print(f"\tCheckpoint granularity: {checkpoint_granularity} blocks.")
+
+
+# Function which stops process after given sleep_time.
+def terminate_process_after(sleep_time: int, checkpoint: int):
+    start = 0.0
+    with aida_log_file.open('r') as f:
+        while True:
+            line = f.readline()
+            if not line:
+                time.sleep(0.1)
+                continue
+            if start > 0 and time.time() - start >= sleep_time:
+                print("Interrupting...")
+                process.terminate()
+                return checkpoint
+            if f"block {checkpoint + checkpoint_granularity}" in line:
+                # First checkpoint was found, we should start timer as it means block processing is running.
+                if start == 0.0:
+                    start = time.time()
+                checkpoint = checkpoint + checkpoint_granularity
+                print(f"Found new checkpoint {checkpoint}.")
+            # If process ends with error (return code 1) or either 'fail' or 'exit status' occurs in line exit script
+            if process.poll() == 1 or any(s in line for s in ["exit status", "fail"]):
+                print("Error occurred - printing output.log:")
+                with open(aida_log_file, 'r') as l:
+                    text = l.read()
+                    print(text)
+                os.chdir(current_dir)
```

Comment:
This line seems to be not part of the responsibility of this function ... changing the current directory in case of an error should probably be added before line 116 and 217, before `sys.exit(1)` is called.